### PR TITLE
cluster/ci/config/prow/config.yaml: Consolidate openshift/origin Tide config

### PR DESCRIPTION
**Pull request by wking**
It was moved to its own section in 070b90a2 (#761) to pick up an `excludedBranches` section.  But that `excludedBranches` section was dropped in 2bd76e3b (#1021).  Now that its Tide config is no longer unique, move it back up into its previous group.

### CODE REVIEW

File: cluster/ci/config/prow/config.yaml [hunk 1]
```diff
@@ -84,15 +85,6 @@ tide:
     - needs-ok-to-test
     - do-not-merge/work-in-progress
     - do-not-merge/hold
-  - repos:
-    - openshift/origin
```
Comment (kargakis) [inline cluster/ci/config/prow/config.yaml]: Please revert this change, we keep origin separate from the rest of the queries to make merge gating on rebases easier.

Comment (wking) [reply cluster/ci/config/prow/config.yaml]: > Please revert this change, we keep origin separate from the rest of the queries to make merge gating on rebases easier.

So something like #707's pulling out openshift/online-registration?  Is there a rule of thumb to decide whether a given repo should get its own section or not?  I'm not clear on why openshift/origin would need a separate section while the rest of the repositories in the previous section do not.

Comment (kargakis) [reply cluster/ci/config/prow/config.yaml]: When @openshift/sig-master want to land a kubernetes rebase in openshift/origin#master, we need to block merges on that branch and at the same time we don't want to be blocking merges elsewhere, hence origin has its own query.
